### PR TITLE
refactor: parse package descriptor in rust

### PR DESCRIPTION
**Pull request by ysndr**
Implements descriptor argument parsing in flox-rust-sdk rather than calling pkgdb.

The new parser is strictly weaker parsing only an attribute path and version, in line with the requirements for catalog based installs and the utilized capabilities of pkgdb.
In particular support for declaring inputs and systems in a single cli argument are dropped.

### CODE REVIEW

File: cli/flox-rust-sdk/src/models/manifest.rs [hunk 1]
```diff
@@ -836,6 +843,94 @@ pub fn temporary_parse_descriptor(descriptor: &str) -> Result<PackageToInstall,
...
+///
+///     attribute_path ::= <install_id> | <attribute_path_rest>.<install_id>
+///     attribute_path_rest ::= <identifier> | <attribute_path_rest>.<identifier>
+///     install_id ::= <identifier>
+///
+///     version ::= <string> # interpreted as semver or plain version by the resolver
+/// ```
+/// Todo: this does currently _not_ handle any more pathological cases like
+///  - `@` in the version string (the last `@` is the delimiter)
+#[derive(Debug, Clone, PartialEq)]
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/manifest.rs:864]:
```suggestion
/// TODO: drop input from PackageToInstall and drop ShorthandDescriptor
#[derive(Debug, Clone, PartialEq)]
```

File: cli/flox-rust-sdk/src/models/manifest.rs [hunk 2]
```diff
@@ -558,7 +558,14 @@ impl FromStr for PackageToInstall {
     type Err = ManifestError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        temporary_parse_descriptor(s)
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/manifest.rs:561]: nonblocking suggestion: drop temporary_parse_descriptor

**Review by mkenigs — APPROVED** [review]

**Comment by mkenigs** [comment]
`input` needs to be removed from unit tests

**Comment by ysndr** [comment]
heh that's new tests that we're not even on this branch yet
should be fixed

**Review by mkenigs — APPROVED** [review]

**Pull request merged**